core/store/src/trie/iterator.rs

```diff
@@ -57,6 +57,22 @@ impl<'a> GenericTrieInternalStorage<TrieStorageNodePtr, ValueHandle> for DiskTri
         Ok(node.unwrap_or_default())
     }
 
+    fn get_node_with_size(
+        &self,
+        ptr: TrieStorageNodePtr,
+        opts: AccessOptions,
+    ) -> Result<GenericTrieNodeWithSize<TrieStorageNodePtr, ValueHandle>, StorageError> {
```

Comment:
nits: Return type `TrieStorageNodeWithSize`. Return `Default::default()` instead of `GenericTrieNodeWithSize::default()`. Change the implementation of `get_node` to call `get_node_with_size` so that we can have a single implementation.